docs/aws/audit/lambdamonitoring/rules/default_timeout_used_remediation.mdx

```diff
@@ -1,6 +1,210 @@
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Lambda service by typing 'Lambda' into the search bar and selecting 'Lambda' from the dropdown menu.
+3. Once you are in the Lambda service, you will see a list of all your Lambda functions. Click on the name of the function you want to check.
+4. In the function's configuration page, look for the 'Basic settings' section. Here, you will see the 'Timeout' setting. If the timeout is not set or is set to a value higher than your organization's standard, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a JSON object with all your Lambda functions.
+
+    ```
+    aws lambda list-functions --region your-region
+    ```
+
+2. Once you have the list of all Lambda functions, you can iterate over each function to get its configuration details. You can use the `aws lambda get-function-configuration` command to get the configuration details of a specific Lambda function.
+
+    ```
+    aws lambda get-function-configuration --function-name your-function-name --region your-region
+    ```
+
+3. In the returned JSON object, look for the `Timeout` attribute. This attribute specifies the amount of time that Lambda allows a function to run before stopping it. If the `Timeout` attribute is not set or is set to a value greater than the recommended maximum, then the function is considered to have a misconfiguration.
+
+4. You can automate the above steps using a Python script. Use the `boto3` library to interact with AWS services. The script will list all Lambda functions, get their configuration details, and check the `Timeout` attribute for each function. If the `Timeout` attribute is not set or is set to a value greater than the recommended maximum, the script will print the function name and its timeout value.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Establish a session with AWS: You need to establish a session with AWS using your access key, secret access key, and the region. Replace 'your_access_key', 'your_secret_access_key', and 'your_region' with your actual AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='your_access_key',
+       aws_secret_access_key='your_secret_access_key',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>